### DISCUSSION TITLE
Feature request: add last_insert_rowid method

### DISCUSSION THREAD

**Discussion by jmatsushita** [rhashimoto/wa-sqlite#151] [General]
Hi there,

I'm giving a shot at implementing the web version of `expo-sqlite` by wrapping the `wa-sqlite` api (with the `crsqlite` extension). [One of the api methods on native](https://github.com/expo/expo/blob/main/packages/expo-sqlite/ios/SQLiteModuleNext.swift#L236) returns the `last_insert_rowid`, however while it seems that the [function is exported](https://github.com/rhashimoto/wa-sqlite/blob/390744d41c61aa0bbd53d3c738abef5e23f71cc4/src/exported_functions.json#L118) it's not currently exposed via the `sqlite-api`.

Would that be of interest? I'm happy to contribute a PR.

Cheers,

Jun

**Comment by rhashimoto** [comment]
This project reserves Issues for project bugs, so feature requests go here in Discussions.

I'm working on major structural changes, so I prefer not to add more things to migrate for now. In this case, since `_sqlite3_last_insert_rowid` is already in the WebAssembly module and doesn't need to convert any types, it should be pretty easy for an application to monkey-patch this in for the time being as you should be able to call `module._sqlite3_last_insert_rowid(db)`.

**Reply by jmatsushita** [reply]
Great thanks for the workaround and good luck on the structural changes. Is there any place on github where I can get a better sense of what these changes will be apart from https://github.com/rhashimoto/wa-sqlite/compare/master...dev ?

**Reply by rhashimoto** [reply]
The [dev branch](https://github.com/rhashimoto/wa-sqlite/tree/dev) is indeed where changes will go. It will be unstable; I don't advise trying to use it yet as I might be rewriting its git commits. I'll be posting updates in [this discussion](https://github.com/rhashimoto/wa-sqlite/discussions/153).
